core/types/schema.go

```diff
@@ -1195,11 +1195,16 @@ func (c *DataType) Clean() error {
 
 // Copy returns a copy of the type.
 func (c *DataType) Copy() *DataType {
-	return &DataType{
-		Name:     c.Name,
-		IsArray:  c.IsArray,
-		Metadata: c.Metadata,
+	d := &DataType{
+		Name:    c.Name,
+		IsArray: c.IsArray,
 	}
+
+	if c.Metadata != nil {
+		d.Metadata = &[2]uint16{c.Metadata[0], c.Metadata[1]}
```

Comment:
It's a pointer because it's optional, but is there a sensible way to keep it a regular `[2]byte` with tests like `c.Metadata == zeroMetaData` or something?

Reply:
We could do that. There are quite a few places where we do `datatype.Metadata == nil`, I guess that would just be `datatype.Metadata == [2]uint26{}`?

Reply:
I can't tell which would be more cumbersome or error prone at a glance.  Whatever feels best to you.

But we'd probably do:

```go
var zeroMetaData [2]uint16 // global

...

if c.Metadata == zeroMetaData {

```

Reply:
The benefit of an actual array field is that it deep copies automatically, and arrays are `comparable`, so you can do simple stuff like `==` there.

If it's optional, then it can be awkward.  Again, totally your call.  Just something to consider.

Reply:
Yeah I can make the change